lib/commands/file-movement.js

```diff
@@ -380,5 +472,21 @@ commands.pullFolder = async function pullFolder (remotePath) {
     : await pullFromRealDevice(this.opts.device, remotePath, false);
 };
 
+/**
+ * @typedef {Object} PullFolderOptions
+ * @property {string} remotePath The full path to the remote folder.
+ */
+
+/**
+ * Pulls the whole folder from the device under test.
+ *
+ * @param {PullFolderOptions} opts
+ * @returns {string} The same as `pullFolder`
+ */
+commands.mobilePullFolder = async function mobilePullFolder (opts = {}) {
+  const { remotePath } = requireArgs('remotePath', opts);
```

Comment:
if rewritten to take advantage of the executeMethodMap, this requirement logic would be taken care of!

Reply:
I just need to investigate code examples better. Could you provide some useful links?